src/workloads/query/GroupLikeDistinct.yml

```diff
@@ -0,0 +1,169 @@
+  $bottomN accumulators so we are including those here in order to see performance changes once that
+  is implemented.
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 2000
+
+Keywords:
+- Distinct
+- Group
+- First
+- Last
+- Top
+- Bottom
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  MaxPhases: &MaxPhases 11
+
+ActorTemplates:
+- TemplateName: RunAggCommand
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "RunAggCommand"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 32
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: {unused: "please specify in which phases this actor should be active."}}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Duration: 30 seconds
+          Collection: *Collection
+          Operations:
+          - OperationMetricsName: DistinctExecutionMetrics
+            OperationName: aggregate
+            OperationCommand:
+              Pipeline: {^Parameter: {Name: "Pipeline", Default: []}}
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: 5000
+        BatchSize: 1000
+        Document:
+          _id: {^Inc: {start: 0, step: 1}}
+          a: {^RandomInt: {min: 0, max: 500}}
+          b: {^RandomInt: {min: 0, max: 500}}
+          c: {^RandomInt: {min: 0, max: 500}}
+        Indexes:
+          - keys: {a: 1, b: 1, c: 1}
+
+# The following use a DISTINCT_SCAN.
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroup"
+      OnlyActiveInPhase: 2
+      Pipeline: [{$group: {_id: "$a"}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSort"
+      OnlyActiveInPhase: 3
+      Pipeline: [{$sort: {a: 1}}, {$group: {_id: "$a"}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortFirst"
+      OnlyActiveInPhase: 4
+      Pipeline: [{$sort: {a: 1, b: 1}}, {$group: {_id: "$a", accum: {$first: "$b"}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortFirstFullDocument"
+      OnlyActiveInPhase: 5
+      Pipeline: [{$sort: {a: 1, b: 1, c: 1}}, {$group: {_id: "$a", accum: {$first: "$$ROOT"}}}]
```

Comment:
Why sort on `c`? It's easier to compare results across different tests if they only vary one parameter at a time. While we mostly analyze the trends for each test separately, when there is a problem, comparing to other tests in the workload might be helpful.